dashboard/lib/projects_list.rb

```diff
@@ -19,13 +19,18 @@ class << self
     # @return [Array<Hash>] An array with each entry representing a project.
     def fetch_section_projects(section)
       [].tap do |projects_list_data|
+        # A hash mapping user IDs to storage IDs.
+        student_storage_ids = PEGASUS_DB[:user_storage_ids].
+          where(user_id: section.students.pluck(:id)).
+          all.
+          map {|h| {h[:user_id] => h[:id]}}.
+          reduce({}, :merge)
         section.students.each do |student|
-          student_storage_id = PEGASUS_DB[:user_storage_ids].where(user_id: student.id).first
-          next unless student_storage_id
-          PEGASUS_DB[:storage_apps].where(storage_id: student_storage_id[:id], state: 'active').each do |project|
+          next unless student_storage_id = student_storage_ids[student.id]
```

Comment:
is `section.students.each` going back to the DB again? if not, it seems better to `raise` if the student storage id doesn't exist, because it would indicate that someone had introduced a programming error.